### PR TITLE
Property type builders, tests

### PR DESCRIPTION
**Pull request by sokolovsv90**
As discussed in pull request #48
All tests passed

### CODE REVIEW

**Comment by borismod** [comment]
@sokolovsv90 are you sure the build passes?

**Comment by sokolovsv90** [comment]
@borismod why do you ask? It does and all the tests pass. I almost feel like I'm missing something

**Comment by borismod** [comment]
For some reason there are $ instead of @  
```C#
if (!validationMap.ContainsKey(validation))
 +                Assert.Inconclusive($"Wrong {nameof(validation)} argument value passed");
 +            validationMap[validation]();
```
http://teamcity.codebetter.com/viewLog.html?buildId=215162&tab=buildResultsDiv&buildTypeId=FluentTc_FluentTcDevelop

image

**Comment by sokolovsv90** [comment]
Oh... theese are special characters for string interpolation introduced in c# 6.0. I certainly can fix that, which version of msbuild do You use?

**Comment by borismod** [comment]
Microsoft Visual Studio 2013

**Comment by sokolovsv90** [comment]
Should work now

**Comment by borismod** [comment]
@sokolovsv90 it get's better:

image

**Comment by sokolovsv90** [comment]
bah, forgot that this was also a feature of c#6 :)

**Comment by sokolovsv90** [comment]
Should work now v.2

File: FluentTc.Tests/Locators/BuildParameterTypeBuilderFormatTests.cs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            if (checkedValue != null)
+                testObject.WithCheckedValue(checkedValue);
+            if (uncheckedValue != null)
+                testObject.WithUncheckedValue(uncheckedValue);
+
+            // Act
+            var result = testObject.Build();
+
+            // Assert
+            Assert.That(result, Is.EqualTo(expected));
```
Comment (borismod) [inline FluentTc.Tests/Locators/BuildParameterTypeBuilderFormatTests.cs]: It's common to use FluentAssertions for verifying test result in FluentTc project

File: FluentTc/Locators/BuildParameterCheckboxTypeBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            m_uncheckedValue = uncheckedValue;
+            return this;
+        }
+
+        public string Build()
+        {
+            var builder = new StringBuilder();
+            builder.Append("checkbox");
+            if (!string.IsNullOrEmpty(m_checkedValue))
+                builder.Append(" checkedValue='").Append(m_checkedValue).Append("'");
```
Comment (borismod) [inline FluentTc/Locators/BuildParameterCheckboxTypeBuilder.cs]: 1. It would be clearer with a single call to AppendFormat can be used instead of several calls to Append. 
2. The format can be a const, like you did in other cases

File: FluentTc/Locators/BuildParameterSelectListTypeBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            if (m_multiple)
+            {
+                builder.Append(" multiple='true'");
+                if (!string.IsNullOrEmpty(m_separator))
+                    builder.Append(" valueSeparator='").Append(m_separator).Append("'");
+            }
+            for (var i = 0; i < m_values.Count; i++)
+            {
+                if (!string.IsNullOrEmpty(m_values[i].Item1))
+                    builder.Append(" label_").Append(i + 1).Append("='").Append(m_values[i].Item1).Append("'");
```
Comment (borismod) [inline FluentTc/Locators/BuildParameterSelectListTypeBuilder.cs]: Same, regarding AppendFormat and const

**Comment by borismod** [comment]
@sokolovsv90 this is a great contribution and this is exactly the way like I would like to see it. just a few minor comments:
1. Replace Append with Append format
2. Use const instead of inline strings
3. Replace Verify.That with FluentAssertions
4. Make sure all new code is covered by unit tests. For example, the following methods are not: BuildParameterTypeBuilder.AsCheckbox and BuildParameterTypeBuilder .AsText
5. Add at least one Acceptance Test for each type: Select, CheckBox, Password, Normal. The below edge case is not unit tested:
image

I am sorry for being so "hard" and I know you did a great job. I'm just trying to keep high standards in this project. I hope you'll understand

**Comment by borismod** [comment]
@sokolovsv90 I assume that I was too hard to contributors. I really appreciate your contribution and I will accept your pull request as is. Can you just resolve the conflicts?

**Comment by sokolovsv90** [comment]
I understand your concerns and totally agree with you in terms of keeping high standards, but I simply haven't noticed how my own stuff got piled up and owerwhelmed me a bit. I'm truly sorry that I couldn't meet all the requirements (and I wish I had time to), but I'm glad you accepted that PR as is, thanks!

**Comment by borismod** [comment]
@sokolovsv90 FluentTc was published to nuget with your contribution, so you're welcome to install its latest package. I would like to thank you for your contribution and will be glad to see you contributing in the future. I'll try to be nicer :smile:

**Comment by borismod** [comment]
@sokolovsv90  what's your full name, so I could give you a credit on contributors list?
